src/autopas/tuning/AutoTuner.cpp

```diff
@@ -280,6 +283,14 @@ void AutoTuner::addMeasurement(long sample, bool neighborListRebuilt) {
 void AutoTuner::bumpIterationCounters() {
   ++_iteration;
   ++_iterationsSinceTuning;
+  if (inTuningPhase()) {
+    ++_iterationsInLastTuningPhase;
+    if (_iterationsInLastTuningPhase > _tuningInterval) {
+      throw autopas::utils::ExceptionHandler::AutoPasException(
+          "Tuning needs more iterations than the tuning interval. Please use a higher value for the tuning interval.");
```

Comment:
Would it be possible to handle this case with a warning instead of an error? Or would this mess with all the counters too much?

Reply:
And how should AutoPas behave in such a case? 
- Restart the tuning? -> This means we'll never converge or test everything.
- Continue the tuning? -> When should then the next tuning phase start?

In my opinion, this might, in theory, be a recoverable failure, but in reality, this is completely undesirable, and I would want to start over with the simulation.

Reply:
I would want continue the tuning.
The relevant case would be a tuning strategy, where the first tuning phase takes significantly more iterations than the subsequent phases.
Say `tuningInterval = 500`, tuning phase 1 takes 510 iterations due to a crazy large search space. Search space for the following tuning phases is reduced such that tuning phase 2 takes only 100 iterations, and tuning phase 3 only 20.

If this is not trivial to handle or totally undesirable, I'm fine to leave it

Reply:
I would be fine with both options really, whatever would be easier to implement. 
Potentially, I like the skipping one tuning phase option more.

My reason to have it as recoverable 'failure', would be to make it more robust to (unexperienced) user input . If you are particularly interested in the tuning process, then this can lead to confusion of course. But then, you should also be able to use sensible values for the tuning interval imo.

Reply:
I think this touches upon the general question how we should treat the case that a tuning phase doesn't start when we expect it. There is also `forceRetune()` with which you can start a tuning phase at any point. If tuning phases are strictly bound to specific iteration intervals then these manually triggered tuning phases can easily run into them.

So let's go for the following:
- Tuning phases always start at `iteration % tuningInterval`, except if we are already in a tuning phase.
- If we are tuning and a new phase is triggered 
  - by the tuning interval simply continue tuning. We can consider this as "merging" two tuning phases.
  - manually via `forceRetune()` abort the current phase and start a new tuning phase. The rationale here is that there must be a reason why we want to start over, e.g. drastic change in the simulation.
- All these cases should throw a meaningful warning.

Reply:
should be implemented now in [01e8856](https://github.com/AutoPas/AutoPas/pull/943/commits/01e8856c60f0e56e2bd2f71935a8f83e2da79c1f)